### DISCUSSION TITLE
Minify/Compress queries

### DISCUSSION THREAD

**Discussion by diaz-hfc** [urql-graphql/urql#3366] [Q&A]
Hey,

I am dealing with a Contentful CMS query limit with is 8KB per query string size and I have to do a very long query, investigating I found that you can compress/minify GraphQL requests, but when I try to do it, looks like urql/core in some way rebuild the query and sends it with lots of /n and empty spaces, increasing the size of my query from 5KB, to 9KB, so I still blocked with this query.

Is there a way I can compress the queries? I was thinking in rewritting the fetchExchange which looks like it is the one is recomposing the query making it big again.

Suggestions?

**Comment by kitten** [comment]
There's some notes on this here which haven't made it into the docs yet: https://github.com/urql-graphql/urql/issues/3073

**Reply by diaz-hfc** [reply]
At the end we started using persistent queries.
